### ISSUE TITLE
Feature request: Apply item to every other page

### ISSUE THREAD

**Issue by aurbus** [ImageProcessing-ElectronicPublications/scantailor-experimental#48]
When one selects the "apply to" button to pick which pages to apply a given effect/option to, other versions of Scantailor have an option to "apply to every other page". This is tremendously helpful if you have a book where, for example, every other page is warped, so you have to apply keystoning/curved lines to every other page, alternating with normal deskewing.

**Comment by zvezdochiot** [comment]
Hi @aurbus .

Provide a small (~50MB) example of a project that requires such a choice at stages 2 and 3 (at other stages such a choice is already available).

**Comment by aurbus** [comment]
Of course...what files/format would you prefer? original PDF (needing processing)? images of pages? 

I have attached the original, unmodified PDF of one such project. If you look closely you can see that roughly every other page is skewed/warped quite a bit to one side, and the opposing pages are more or less fine, maybe having some keystoning occasionally. I could apply 'curved lines' to every page, but I find it to be more temperamental than simple deskew or keystoning, which makes the processing take much longer in such cases. 

Happy to provide more examples if it helps...

**Comment by zvezdochiot** [comment]
@noobie-iv , ты в эскизах ковырялся. Есть идеи добавления в меню эскизов функцаналала выбора страниц? Типа: "выбрать всё", "выбрать чётные", "выбрать нечётные", "инвертировать"? То, как сейчас работает через "Применить к ..." - это не дело! Пользователь не видит выбора и не может им манипулироваль. Лажа то бишь. Такие вот дела.

**Comment by plzombie** [comment]
@zvezdochiot А разве сейчас нет такой кнопки?
![изображение](https://github.com/user-attachments/assets/4542cca3-bb1b-4eaa-9394-cd1ce58f9c2f)
![изображение](https://github.com/user-attachments/assets/307ed1ce-b81b-499b-bae3-c60a578cc5d4)

**Comment by zvezdochiot** [comment]
@plzombie , так в том то и дело. Только видишь, что народ просит: не хватает где то "применить через одну". Потом ещё попрося что то. И при этом ты не видишь к чему применяется. От того я и говорю, что должно быть просто "Применить к выбранному" без всяких диалогов, а сам выбор должен быть в эскизах: "выбрать всё", "выбрать после", ... Чтобы было видно, к чему именно применяется и в случае необходимости можно было это подправить - изменить выбор через "Ctrl".

**Comment by plzombie** [comment]
@zvezdochiot Так там как раз где превьюхи выбираешь, зажимая ctrl или shift + лкм. Потом уже жмёшь применить

**Comment by zvezdochiot** [comment]
@plzombie ну правильно. Только в превьюшках нет в меню "выбрать всё", "выбрать после", "выбрать через одну",... А ежели бы было, то эти дурацкие диалоговые окна нафиг не нужны, просто смотришь что выбрано и жмёшь "Применить к выбранному", и готово.

**Comment by noobie-iv** [comment]
@zvezdochiot 
> ты в эскизах ковырялся.

Ковырялся. На них я и приостановился несколько месяцев назад. Видел какие-то контекстные менюшки в ThumbnailSequence. Но в минимальный проект не включал. Вообще не понимаю, зачем нужен этот дополнительный класс, который хранит указатели на эскизы одновременно с гуевым окном, где они отображаются, вместе с 50k кода по их синхронизации. В книжках по Qt вроде просто подсовывают окну класс-модель, которая по запросу должна возвращать картинку на показ; по книжкам это должна была быть обертка над кешем. Заодно эта последовательность тянет за собой boost::multi_index_container, который в mvst удалось вырезать; второй такой же сидит в кеше эскизов (из-за них в ветке по удалению буста проблемы будут).

> Есть идеи добавления в меню эскизов функцаналала выбора страниц?

Это старый вопрос. Из-за которого я полез код копипастить в STU, а не в STEX. Мне казалось, что копипасты из STEX в STU будет меньше, чем из STU в STEX. Потому что STEX - очень старая ветка, и в STU/STA добавлено уже столько, что забирать фичи оттуда - на много лет работы.

А оказалось, что и просто в коде разобраться небыстро, уже год попыток с околонулевым результатом. За этот год у нас в конторе купили техэксперт, и еженедельная обработка нормативов у меня отвалилась, я больше не активный пользователь ST. Хотя мне все еще кажется, что осталось чуть-чуть, так что какое-то время я еще тут. Но до завершения MVST я все равно ни с чем помочь не смогу. Любые вмешательства в гуй требуют знать, как он таки работает. Надеюсь, после MVST сваяю диаграмки связей/сигналов/событий, по которым можно будет вмешиваться в код по-большому.
